README.md

```diff
@@ -76,6 +79,169 @@ INFO MKE cluster admin UI: https://test-mke-cluster.example.com
+    #     port = 22
+    #   }
+    # }
+    # host {
+    #   role = "worker"
+    #   winrm {
+    #     address = "win1.example.org"
+    #     user = "Administrator"
+    #     password = "my-admin-password"
+    #   }
+    # }
+    # ```
+    #
+    # Below we use terraform dynamid blocks to iterate
+    # across a list of machine maps which each have a
+    # role, and either ssh or winrm connection details.
+    #
+
+    dynamic "host" {
+      # dynamic host blocks built from the hosts map
+      # provided by the module as an output:
+      #
+      # module.provision.hosts = [
+      #  {
+      #    role = "manager"
+      #    ssh = {
+      #      address = "manager.example.org"
+      #      user = "ubuntu"
+      #      key_path = "./ssh-key.pem"
+      #      port = 22
+      #    }
+      #  },
+      #  {
+      #    role = "worker"
+      #    winrm {
+      #      address = "win1.example.org"
+      #      user = "Administrator"
+      #      password = "my-admin-password"
+      #    }
+      #  },
+      #  ...
+      # ]
+      #
+      for_each = module.provision.hosts
+
+      content {
+        role = host.value.role
+
+        # If the host map has ssh settings
+        dynamic "ssh" {
+          for_each = can(host.value.ssh) ? [1] : [] # one loop if there er a value
+
+          content {
+            address  = host.value.ssh.address
+            user     = host.value.ssh.user
+            key_path = host.value.ssh.keyPath
+            port     = 22
+          }
+        }
+
+        # If the host map has ssh settings
+        dynamic "winrm" {
+          for_each = can(host.value.winRM) ? [1] : [] # one loop if there er a value
+
+          content {
+            address   = host.value.winRM.address
+            user      = host.value.winRM.user
+            password  = host.value.winRM.password
+            use_https = host.value.winRM.useHTTPS
+            insecure  = host.value.winRM.insecure
+            port      = 5985
+          }
+        }
+
+      }
+    }
+
+    # MCR configuration
+    mcr {
+      channel             = "stable"
+      install_url_linux   = "https://get.mirantis.com/"
+      install_url_windows = "https://get.mirantis.com/install.ps1"
+      repo_url            = "https://repos.mirantis.com"
+      version             = var.mcr_version
+    } // mcr
+
+    # MKE configuration
+    mke {
+      admin_password = var.mke_password
+      admin_username = "admin"
+      image_repo     = "docker.io/mirantis"
+      version        = var.mke_version
+      install_flags  = ["--san=${module.provision.mke_lb}", "--default-node-orchestrator=kubernetes", "--nodeport-range=32768-35535"]
+      upgrade_flags  = ["--force-recent-backup", "--force-minimums"]
+    } // mke
+
+    # MSR configuration
+
+    msr {
+      image_repo    = "docker.io/mirantis"
+      version       = var.msr_version
+      replica_ids   = "sequential"
+      install_flags = ["--ucp-insecure-tls"]
+    } // msr
```

Comment:
currently, putting in an msr block, without any msr hosts, will fail in the launchpad provider with a descriptive error.  In your example here, you have `msr_hosts=0`, so it will fail.

Reply:
Setting the example to have msr
`msr_hosts=1`